[Shipping Labels] New UI for shipments in Order Details

hichamboushaba opened the pull request:
<!-- Remember about a good descriptive title. -->

<!-- Id number of the Linear issue this PR addresses, e.g., WOOMOB-373. -->
Part of WOOMOB-972

### Description
This PR adds a new section to order details for displaying the Woo Shipping shipments and their statuses, it shows:
- Non-purchased shipments with their items, and allows starting shipping labels creation for them.
- Purchased shipments with their items, tracking number, and allows viewing more details.

To handle the correct display of the list in all conditions, the following changes were applied to the order details screen:
- The legacy "Create shipping labels" button is shown only if the order is using the legacy shipping labels plugin.
- The legacy shipping labels list is shown only when the legacy plugin is used.
- A new property `wooShippingShipmentsList` is used for the new plugin.

> [!IMPORTANT]
> The refund notice (bsLNrhmi2xQZB4C0TzjmHB-fi-4490_67707) is not implemented yet, it will require updating `GetShipments` logic to make sure we preserve the label when refuned, I will handle it in a separate PR. 

### Steps to reproduce
##### Ineligible store
1. Update the store address to a location outside of US.
2. Open an order.
3. Confirm the shipments list is not shown.

##### Eligible store with order that has no shipments
1. Make sure the store is eligible.
2. Create a new order.
3. Open the order in the app.
4. Confirm that the shipments section is shown, with a single shipment that lists all products.
5. Tap on Create shipping label and confirm it opens the correct screen.

##### Eligible store with order that has fulfilled shipments
1. Make sure the store is eligible.
2. Open an order that has fulfilled shipments.
3. Confirm that the shipments section is shown, and shows the fulfilled shipments.
4. Check the menus and confirm they work as expected.
5. Tap on "View shipping label" and confirm it works.

##### Store with legacy plugin
1. Disable the new plugin and if neede install the legacy plugin
2. Re-open the app (to make sure plugins are refetched)
3. Check an order that has some legacy plugins (or create new ones)
4. Confirm they are displayed correctly.

### Testing information
- Confirm the new section looks like the design, and works correctly.
- Confirm there is no regression for the legacy plugin.

### The tests that have been performed
The above.

### Images/gif
<img width="360" alt="Screenshot_20250819_181500" src="https://github.com/user-attachments/assets/2977fbcd-cdbb-43e7-9e04-110517edc3f9" />
<img width="360" alt="Screenshot_20250819_181639" src="https://github.com/user-attachments/assets/2a8b38c0-e0eb-4fda-8517-e511d1a1c52c" />
<img width="360" alt="Screenshot_20250819_181634" src="https://github.com/user-attachments/assets/9bc08543-dc16-42b3-a1d2-106b51f13802" />
<img width="360" alt="Screenshot_20250819_181513" src="https://github.com/user-attachments/assets/c37c2802-b71d-43f3-9d73-61117bca1b05" />


- [ ] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

hichamboushaba committed:
Add logic to fetch Woo Shipping shipments in Order Details

hichamboushaba committed:
Remove unused argument

And fix an issue with showing/hiding the shipping labels

hichamboushaba committed:
Start implementing shipment details view in Order Details

hichamboushaba committed:
Add items row to the shipment details

hichamboushaba committed:
Add tracking number row

hichamboushaba committed:
Handle tracking actions

hichamboushaba committed:
Hide track action if the tracking link is empty

hichamboushaba committed:
Handle main click events

hichamboushaba committed:
Improve paddings

hichamboushaba committed:
Handle refund request

dangermattic commented:
<!--
  0 Errors
  2 Warnings: This PR is larger than 300 lin...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/265">23.1</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

hichamboushaba committed:
Hide shipping labels creation button when using new Woo Shipping plugin

The new Woo Shipping section in order details allows starting shipping labels creation.

hichamboushaba committed:
Hide the woo shipping section when not eligible

hichamboushaba committed:
Fix logic of shipment sections header

hichamboushaba committed:
Fix and add unit tests

hichamboushaba committed:
Set default value for containsAddonsFrom mock

And remove all not-needed mocking calls

hichamboushaba committed:
Remove unused code

hichamboushaba committed:
Fix detekt issues

hichamboushaba commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/details/OrderDetailViewModel.kt:
```diff
@@ -904,26 +911,37 @@ class OrderDetailViewModel @Inject constructor(
     }
 
+    private suspend fun loadWooShippingShipments(): ListInfo<ShipmentUIModel> {
+        if (!isRevampWooShippingEnabled ||
+            eligibilityDataStore.observeEligibility(navArgs.orderId).first() != true
+        ) {
+            return ListInfo(isVisible = false)
+        }
```
When the order is not eligible for shipping labels creation, we completely hide the section, which means we don't show even the created labels.

I opted for this behavior because when viewing a purchased label, we will allow the user to create other labels if there are unfulfilled shipments, and changing this behavior will take more effort, and IMO it's not worth it for this edge case, I notified the iOS team about this behavior change here (p1755620365086799-slack-C03L1NF1EA3).

hichamboushaba commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/details/OrderDetailViewModel.kt:
```diff
@@ -904,26 +911,37 @@ class OrderDetailViewModel @Inject constructor(
         val orderRefunds = loadOrderRefunds()
         val orderProducts = loadOrderProducts(orderRefunds)
 
-        if (shippingLabels.isVisible) {
-            _shippingLabels.value = shippingLabels.list
-        }
```
This logic here wasn't always working as expected, as if for some reason we needed to hide the section on a second call to `displayProductAndShippingDetails`, we wouldn't do it, because we were updating the list only when `isVisible` is `true`.
I updated the logic in line 944.

hichamboushaba commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/details/OrderDetailViewState.kt:
```diff
@@ -16,7 +16,6 @@ data class OrderDetailViewState(
     val refreshedProductId: Long? = null,
     val isCreateShippingLabelButtonVisible: Boolean? = null,
     val isProductListVisible: Boolean? = null,
-    val areShippingLabelsVisible: Boolean? = null,
```
This wasn't used anywhere.

hichamboushaba committed:
Update style of the view label button

hichamboushaba committed:
Update Material Theme 2 dark colors

Use various surface containers colors to replicate the overlay behavior of Material 2

hichamboushaba committed:
Update background of the shipments list to match other order details sections

hichamboushaba commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/details/views/OrderDetailWooShippingShipmentListView.kt:
```diff
@@ -0,0 +1,347 @@
+        )
+    }
+
+    if (isDialogShown) {
+        Dialog(onDismissRequest = { isDialogShown = false }) {
```
For displaying the shipment items, I opted for using a simple dialog, as this is quite simpler, and IMO it offers the same experience. Please feel free to share your thoughts about this.

hichamboushaba committed:
Add release note

hichamboushaba commented on WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/OrderDetailViewModelTest.kt:
```diff
@@ -117,7 +119,9 @@ class OrderDetailViewModelTest : BaseUnitTest() {
         onBlocking { getOrderDetailsPluginsInfo() } doReturn pluginsInfo
     }
-    private val addonsRepository: AddonRepository = mock()
+    private val addonsRepository: AddonRepository = mock {
+        onBlocking { containsAddonsFrom(any()) } doReturn false
+    }
```
This test class needs some refactoring to make it easier to add new tests. For this PR, I just modified this part, I added this default mock to avoid requiring mocking the repository response for all tests.

hichamboushaba commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/compose/theme/WooColors.kt:
```diff
@@ -111,9 +111,9 @@ val Material3DarkColorScheme
-        surfaceContainer = colorResource(R.color.color_surface_elevated),
+        surfaceContainer = colorResource(R.color.color_surface_elevated_04),
         surfaceContainerHigh = colorResource(R.color.color_surface_elevated),
         surfaceContainerHighest = colorResource(R.color.color_surface_elevated),
-        surfaceContainerLow = colorResource(R.color.color_surface_elevated),
-        surfaceContainerLowest = colorResource(R.color.color_surface_elevated),
+        surfaceContainerLow = colorResource(R.color.color_elevation_overlay_01),
+        surfaceContainerLowest = colorResource(R.color.color_elevation_overlay_01),
```
Using `color_surface_elevated` for all `surfaceContainer` variants didn't match what we use in Material 2, using the following colors seems to work well (at least for now). I hope we'll get a chance to work on a full color Material 3 color palette soon.

JorgeMucientes replied:
Hey @hichamboushaba  heads up that for now, I'll be reverting these color changes. This resulted in bottom sheets and drop-down menus displaying transparent [background in dark mode](https://linear.app/a8c/issue/WOOMOB-1190/transparent-backgrounds-in-dark-mode-on-android).  

I'm unsure what the prior issue was when using `color_surface_elevated`. Could you share a screenshot of it 🙏🏼

hichamboushaba committed:
Merge branch 'trunk' into issue/WOOMOB-972-new-SL-order-details-UI

irfano commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/details/adapter/OrderDetailShippingLabelsAdapter.kt:
```diff
@@ -249,44 +241,25 @@ class OrderDetailShippingLabelsAdapter(
-                viewBinding.shippingLabelItemViewMoreButtonTitle.isVisible = false
+            // Legacy flow buttons
+            viewBinding.shippingLabelItemViewMore.isVisible = true
+            viewBinding.shippingLabelItemViewMoreButtonTitle.isVisible = true
```
Since these views are visible by default, we can remove these lines.

hichamboushaba replied:
2f8d214

irfano reviewed (changes requested):
Great job! PR looks good, I'll just add minor comments.

**Question:** Did you get this item dialog design from Figma? I couldn’t find it.
<img width="300" alt="Screenshot_20250820_220907" src="https://github.com/user-attachments/assets/31e1ab55-d40f-4b8c-a230-27010380da6e" />

- np: When fonts are larger, `>` icon at the end of rows is not vertically centered.
- We can add support for the singular case of 1 item. Currently, it's displayed as `1 items`.
<img width="400" alt="Screenshot_20250820_221246" src="https://github.com/user-attachments/assets/1890e430-5ee9-4dbc-bba5-cb90d386a6a2" />

hichamboushaba committed:
Make sure chevrons are centered vertically

hichamboushaba committed:
Use pluralized value for items

hichamboushaba committed:
Remove redundant code

hichamboushaba commented:
Thanks @irfano for the review.

> Question: Did you get this item dialog design from Figma? I couldn’t find it.

I left a comment about it [here](https://github.com/woocommerce/woocommerce-android/pull/14485#discussion_r2285934503), and to answer your question, Figma doesn't have a design, iOS app used the same screen of the Refund Details:
<img width="360" alt="Simulator Screenshot - iPhone 16 - 2025-08-21 at 10 00 27" src="https://github.com/user-attachments/assets/8813802b-073a-40d5-b873-9bbf673ca6b4" />

This was simpler for them because the screen was already reusable, for us I had the choice between working on a new screen or using the dialog, and I opted for the dialog. Do you have any remarks about the dialog?

irfano reviewed (approved):
> I left a comment about it [here](https://github.com/woocommerce/woocommerce-android/pull/14485#discussion_r2285934503)

Thanks, I missed that. The dialog looks good and works well in dark, landscape, and accessibility modes.

LGTM! 👍🏻

hichamboushaba committed:
Merge branch 'trunk' into issue/WOOMOB-972-new-SL-order-details-UI
